From f42ccd68d0606def95789e179cd447420c7e20b3 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Wed, 26 Jun 2013 17:16:29 +0300
Subject: [PATCH 058/429] dpaa_eth: removed empty ndo callbacks

Empty callbacks are not required.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: Ia41366d99c455b976bb0d4376a5fdc63332c58f2
Reviewed-on: http://git.am.freescale.net:8181/3094
Reviewed-by: Sovaiala Cristian-Constantin-B39531 <Cristian.Sovaiala@freescale.com>
Reviewed-by: Radulescu Ruxandra Ioana-B05472 <ruxandra.radulescu@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[origin patch is from QorIQ-SDK-V1.5-20131219-yocto]
Signed-off-by: Yang Wei <Wei.Yang@windriver.com>
---
 .../net/ethernet/freescale/dpa/dpaa_eth_macless.c  |   14 --------------
 1 files changed, 0 insertions(+), 14 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
index d4610c9..8d3621e 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_macless.c
@@ -107,11 +107,8 @@ static int __cold dpa_macless_start(struct net_device *net_dev);
 static int __cold dpa_macless_stop(struct net_device *net_dev);
 static int dpa_set_macless_address(struct net_device *net_dev, void *addr);
 static int dpaa_eth_macless_probe(struct platform_device *_of_dev);
-static void dpa_set_macless_rx_mode(struct net_device *net_dev);
 static netdev_features_t
 dpa_macless_fix_features(struct net_device *dev, netdev_features_t features);
-static int dpa_macless_ioctl(struct net_device *dev,
-			     struct ifreq *rq, int cmd);
 static int dpa_macless_netdev_init(struct device_node *dpa_node,
 				struct net_device *net_dev);
 
@@ -127,11 +124,9 @@ static const struct net_device_ops dpa_macless_ops = {
 	.ndo_select_queue = dpa_select_queue,
 #endif
 	.ndo_change_mtu = dpa_change_mtu,
-	.ndo_set_rx_mode = dpa_set_macless_rx_mode,
 	.ndo_init = dpa_ndo_init,
 	.ndo_set_features = dpa_set_features,
 	.ndo_fix_features = dpa_macless_fix_features,
-	.ndo_do_ioctl = dpa_macless_ioctl,
 };
 
 static const struct of_device_id dpa_macless_match[] = {
@@ -171,10 +166,6 @@ static int __cold dpa_macless_stop(struct net_device *net_dev)
 	return 0;
 }
 
-static void dpa_set_macless_rx_mode(struct net_device *net_dev)
-{
-}
-
 static int dpa_set_macless_address(struct net_device *net_dev, void *addr)
 {
 	const struct dpa_priv_s	*priv;
@@ -212,11 +203,6 @@ dpa_macless_fix_features(struct net_device *dev, netdev_features_t features)
 	return features;
 }
 
-static int dpa_macless_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
-{
-	return -EINVAL;
-}
-
 static int dpa_macless_netdev_init(struct device_node *dpa_node,
 				struct net_device *net_dev)
 {
-- 
1.7.5.4

